docs/products/kafka/howto/ksql-docker.rst

```diff
@@ -0,0 +1,106 @@
+
+To have ksqlDB working with Aiven's `Karapace <https://karapace.io/>`__ Schema Registry you need to explicitly define a truststore that contains the commonly trusted root CA of Schema Registry server. To create such a truststore:
+
+1. Obtain the root CA of the server with the following ``openssl`` command by replacing the ``APACHE_KAFKA_HOST`` and ``SCHEMA_REGISTRY_PORT`` placeholders::
+
+    openssl s_client -connect APACHE_KAFKA_HOST:SCHEMA_REGISTRY_PORT \
+        -showcerts < /dev/null 2>/dev/null | \
+        awk '/BEGIN CERT/{s=1}; s{t=t "\n" $0};
+            /END CERT/ {last=t; t=""; s=0}; END{print last}' \
+        > ca_schema_registry.cert
+
+2. Create the truststore with the following ``keytool`` command  by replacing the ``TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME`` placeholder::
+
+    keytool -import -file ca_schema_registry.cert \
+        -alias CA \
+        -keystore \
+        TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME
+   
+   You'll be prompted to insert a password, which in this example is referenced by the placeholder ``TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD``
+
+.. Tip::
+
+    The ``TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME`` can be any name with extension should be ``.jks``
+
+
+Run ksqlDB on Docker
+--------------------
+
+You can run ksqlDB on Docker with the following command, by replacing the placeholders:
+
+* ``SSL_STORE_FOLDER``
+* ``APACHE_KAFKA_HOST``
+* ``APACHE_KAFKA_PORT``
+* ``KEYSTORE_FILE_NAME``
+* ``SSL_KEYSTORE_PASSWORD``
+* ``SSL_KEY_PASSWORD``
+* ``TRUSTSTORE_FILE_NAME``
+* ``SSL_TRUSTSTORE_PASSWORD``
+* ``SCHEMA_REGISTRY_PORT``
+* ``SCHEMA_REGISTRY_PASSWORD``
+* ``TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME``
+* ``TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD``
+
+::
+
+    docker run -d --name ksql  \
+        -v SSL_STORE_FOLDER/:/ssl_settings/ \
+        -p 127.0.0.1:8088:8088 \
+        -e KSQL_BOOTSTRAP_SERVERS=APACHE_KAFKA_HOST:APACHE_KAFKA_PORT \
+        -e KSQL_LISTENERS=http://0.0.0.0:8088/ \
+        -e KSQL_KSQL_SERVICE_ID=ksql_service_1_ \
+        -e KSQL_OPTS="-Dsecurity.protocol=SSL
+            -Dssl.keystore.type=PKCS12
+            -Dssl.keystore.location=/ssl_settings/KEYSTORE_FILE_NAME
+            -Dssl.keystore.password=SSL_KEYSTORE_PASSWORD
+            -Dssl.key.password=SSL_KEY_PASSWORD
+            -Dssl.truststore.type=JKS
+            -Dssl.truststore.location=/ssl_settings/TRUSTSTORE_FILE_NAME
+            -Dssl.truststore.password=SSL_TRUSTSTORE_PASSWORD
+            -Dksql.schema.registry.url=APACHE_KAFKA_HOST:SCHEMA_REGISTRY_PORT
+            -Dksql.schema.registry.basic.auth.credentials.source=USER_INFO
+            -Dksql.schema.registry.basic.auth.user.info=avnadmin:SCHEMA_REGISTRY_PASSWORD
+            -Dksql.schema.registry.ssl.truststore.location=/ssl_settings/TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME
+            -Dksql.schema.registry.ssl.truststore.password=TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD" \
+        confluentinc/ksqldb-server:0.23.1
+
```

Comment:
Some docker setup has issues using mounting options `-v`
In my case using `minikube + hyperkit + dockercli` it's a pain... Perhaps provide alternative option here just copy the keystores into the container?

```
docker container create --name ksql  \
    -p 127.0.0.1:8088:8088 \
    -e KSQL_BOOTSTRAP_SERVERS=APACHE_KAFKA_HOST:APACHE_KAFKA_PORT \
    -e KSQL_LISTENERS=http://0.0.0.0:8088/ \
    -e KSQL_KSQL_SERVICE_ID=ksql_service_1_ \
    -e KSQL_OPTS="-Dsecurity.protocol=SSL
        -Dssl.keystore.type=PKCS12
        -Dssl.keystore.location=/home/appuser/KEYSTORE_FILE_NAME
        -Dssl.keystore.password=SSL_KEYSTORE_PASSWORD
        -Dssl.key.password=SSL_KEY_PASSWORD
        -Dssl.truststore.type=JKS
        -Dssl.truststore.location=/home/appuser/TRUSTSTORE_FILE_NAME
        -Dssl.truststore.password=SSL_TRUSTSTORE_PASSWORD
        -Dksql.schema.registry.url=APACHE_KAFKA_HOST:SCHEMA_REGISTRY_PORT
        -Dksql.schema.registry.basic.auth.credentials.source=USER_INFO
        -Dksql.schema.registry.basic.auth.user.info=avnadmin:SCHEMA_REGISTRY_PASSWORD
        -Dksql.schema.registry.ssl.truststore.location=/home/appuser/TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME
        -Dksql.schema.registry.ssl.truststore.password=TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD" \
    confluentinc/ksqldb-server:0.23.1
docker cp KEYSTORE_FILE_NAME ksql:/home/appuser/
docker cp TRUSTSTORE_FILE_NAME ksql:/home/appuser/
docker cp TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME ksql:/home/appuser/
docker start ksql
```